### PR TITLE
[Docs] Docs/update install

### PR DESCRIPTION
**Pull request by jamessouth**
Hello, this is my re-write of the install documentation in conjunction with my [PR from the main repo](https://github.com/LunarVim/LunarVim/pull/1236). I have focused mainly on the `Getting Started` section but also made some changes to the `For Devs` section - I moved the `Compile Neovim from Source` and `Manual Install` sections in (presumed) accordance with [this commit from today](https://github.com/LunarVim/lunarvim.org/commit/93ce5873152a3e457511b1b901bcf9f0d9f2ceca). I have mostly copied the `Manual Install` instructions and I have begun to align them with my install script but it looks like more changes will be necessary. I have not yet tested a manual install. As with my install script PR, I'm not sure how to handle Gentoo.

Fixes numerous punctuation and grammatical errors/inconsistencies, broken links, etc. Hope this helps!

### CODE REVIEW

**Comment by rebuilt** [comment]
Thank you for your work on this.  We are currently rethinking the structure for the documentation.  We want to make getting started as simple as possible by removing the optional or advanced features from the main navigation.  Chris went through the docs to slim down the verbosity, giving preference to code snippets over explanatory paragraphs.  He placed the ` advanced ` documentation  in `docs/drafts` until we finalize a structure for that information.  I think the manual install procedures belong in that advanced section.  Going forward, the less a user has to read, the better.

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ module.exports = {
         link: "/sponsors/",
       },
     ],
-    repo: "https://github.com/ChristianChiarulli/LunarVim",
+    repo: "https://github.com/LunarVim/LunarVim",
```
Comment (rebuilt) [inline docs/.vuepress/config.js:44]: This is important.  Can you move this to it's own pr and I'll accept it.

**Review by rebuilt — CHANGES_REQUESTED** [review]
Chris intentionally removed a lot of this information so I can't just add it back in.  The best course of action is to add the relevant bits to `docs/drafts` so we can rework it later.  

The correction of the github repo name is important so please create a separate commit for that.

File: docs/.vuepress/config.js (hunk 1 above)
Comment (jamessouth) [reply docs/.vuepress/config.js:44]: So to fix this would I `git checkout <this commit> this/file/path`, change it back and push to this branch?

**Comment by jamessouth** [comment]
I think just closing this would be best and I'll come back with everything moved to drafts.

**Pull request closed without merging**